### DISCUSSION TITLE
Setting Preview image with api

### DISCUSSION THREAD

**Discussion by wanzenried** [Part-DB/Part-DB-server#529] [Q&A] [answered]
Using the patch part api, i managed to set an image as an attachment for a file. Is it possible to set the preview image of the the part using the api aswell?

Seeing as the difference between parts with preview on / off is the existence of "master_picture_attachment" when getting the data of the part using the api, i tried setting that using the part patch api:

"master_picture_attachment": [
        {
            "name":"test",
            "attachment_type": "/api/attachment_types/1",
            "url":"http://localhost:8080/media/test/test.jpg"
        } 
    ]

But doing this gives me the following error:

"propertyPath": "master_picture_attachment",
         "message": "The preview attachment must be a valid picture!",
            "code": "6b3befbc-2f01-4ddf-be21-b57898905284"

What am i doing wrong here?

**Comment by jbtronics** [comment, marked as answer]
Working with attachments using API is currently quite limited (see issue #502 ).

Also the preview picture should be one of the already existing attachments as part. So you should  pass an IRI of an already existing attachment to the master_picture_attachment.

**Comment by jbtronics** [comment]
With Part-DB v1.11 it is possible to properly work with attachments via the API. You can also directly upload files or let Part-DB download files.
